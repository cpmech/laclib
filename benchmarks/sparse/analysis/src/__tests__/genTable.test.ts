import { genTable } from '../genTable';
import { IReport, zeroReport } from '../types';

describe('genTable', () => {
  it('generates default table', () => {
    const r = zeroReport();
    r.StepAnalyze.ElapsedTimeString = '1s';
    r.StepFactorize.ElapsedTimeString = '2s';
    r.StepSolve.ElapsedTimeString = '3s';
    r.TimeSolverString = '6s';
    const reports: IReport[] = [r, r, r, r];
    const table = genTable('mytable', reports);
    expect(table).toMatchSnapshot();
  });

  it('generates OMP table', () => {
    const r = zeroReport();
    r.StepAnalyze.ElapsedTimeString = '1s';
    r.StepFactorize.ElapsedTimeString = '2s';
    r.StepSolve.ElapsedTimeString = '3s';
    r.TimeSolverString = '6s';
    const reports: IReport[] = [r, r, r, r];
    const table = genTable('mytable', reports, true);
    expect(table).toMatchSnapshot();
  });
});
