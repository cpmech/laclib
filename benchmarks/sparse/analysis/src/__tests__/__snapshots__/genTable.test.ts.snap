// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genTable generates OMP table 1`] = `
"<table style=\\"border: 1px solid #979797; border-collapse: collapse;\\">
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">mytable</td>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">OMP nt 1</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">OMP nt 2</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">OMP nt 3</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">OMP nt 4</th>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Analyze</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1s</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Factorize</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2s</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Solve</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3s</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Total</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6s</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Linf(A.x)</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Rel Error</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0.00e+0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0.00e+0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0.00e+0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0.00e+0</td>
  </tr>
</table>
"
`;

exports[`genTable generates default table 1`] = `
"<table style=\\"border: 1px solid #979797; border-collapse: collapse;\\">
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">mytable</td>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">MPI np 1</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">MPI np 2</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">MPI np 3</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">MPI np 4</th>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Analyze</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1s</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Factorize</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2s</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Solve</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3s</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Total</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6s</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6s</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Linf(A.x)</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">Rel Error</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0.00e+0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0.00e+0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0.00e+0</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">0.00e+0</td>
  </tr>
</table>
"
`;
